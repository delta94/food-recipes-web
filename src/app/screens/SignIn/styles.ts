import styled, { keyframes } from 'styled-components'

import bg from '../../../assets/img/bg--signin-signup.png'

export const Container = styled.div`
  height: 100vh;

  display: flex;
  align-items: stretch;
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  place-content: center;
  width: 100%;
  max-width: 700px;
`

const appearFromLeft = keyframes`
  from {
    opacity: 0;
    transform: translateX(-50px);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
`

export const AnimationContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;

  animation: ${appearFromLeft} 1s;

  img {
      max-width: 200px;
  }

  form {
    margin: 80px 0;
    width: 100%;
    max-width: 340px;
    text-align: center;
    display: flex;
    flex-direction: column;

    input {
        height: 2.8rem;
        border: 0;
        padding-left: 1rem;
        border-radius: 8px;
        margin-top: 1.5rem;
        border: 1px solid lightgrey;
    }

    button {
        height: 2.5rem;
        margin-top: 1.5rem;
        border: 0;
        border: 1px solid lightgrey;
    }

    h1 {
      margin-bottom: 24px;
      color: #312e2d;
    }

    a {
      color: #312e2d;
      display: block;
      margin-top: 24px;
      text-decoration: none;
      transition: color 0.2s;

    }
  }

  > a {
    color: #312e2d;
    display: block;
    margin-top: 24px;
    text-decoration: none;
    transition: color 0.2s;

    display: flex;
    align-items: center;
  }
`

export const Background = styled.div`
  flex: 1;
  background: url('${bg}') no-repeat center;
  background-size: cover;
`
