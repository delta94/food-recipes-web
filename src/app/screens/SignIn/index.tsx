import React, { ChangeEvent, useState } from 'react'
import { useDispatch } from 'react-redux'
import { Link } from 'react-router-dom'
import logo from '../../../assets/img/logo.png'
import { Creators } from '../../../core/redux/store/ducks/auth'
import { toast } from 'react-toastify'
import * as Yup from 'yup'

import { Container, Content, AnimationContainer, Background } from './styles'

const schema = Yup.object().shape({
  email: Yup.string()
    .email('Insira um e-mail válido')
    .required('O email é obrigatório'),
  password: Yup.string().required('A senha é obrigatória')
})

const SignIn = () => {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')

  const dispatch = useDispatch()

  async function handleSubmit (event: ChangeEvent<HTMLFormElement>) {
    event.preventDefault()

    try {
      // await schema.validate(schema)

      dispatch(Creators.signInRequest(email, password))
    } catch (error) {
      toast(error.message)
    }

    // dispatch(Creators.login('pass', 'asasa'))
  }

  return (
    <Container>
      <Content>
        <AnimationContainer>
          <Link to="/"><img src={logo} alt="Recipes" /></Link>

          <form onSubmit={handleSubmit}>
            <h1>Faça seu login</h1>

            <input name="email" placeholder="E-mail" value={email} onChange={(e) => setEmail(e.target.value)} />

            <input
              name="password"
              type="password"
              placeholder="Senha"
              value={password} onChange={(e) => setPassword(e.target.value)}
            />

            <button type="submit">Entrar</button>
          </form>

          <Link to="/signup">
            Criar conta
          </Link>
        </AnimationContainer>
      </Content>
      <Background />
    </Container>
  )
}

export default SignIn
