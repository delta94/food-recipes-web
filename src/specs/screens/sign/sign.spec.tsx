import React from 'react'
import { act, fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import * as redux from 'react-redux'

import SignIn from '../../../app/screens/SignIn'

import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

const mockStore = configureStore([])

describe('[SCREENS] sign', () => {
  let store: any

  beforeEach(() => {
    store = mockStore({
      auth: {}
    })
  })

  it('Should render correctly', () => {
    const container = render(
      <BrowserRouter>
        <Provider store={store}>
          <SignIn />
        </Provider>
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('Should return if onSubmit was called successfully', () => {
    const { getByText, getByPlaceholderText } = render(
      <BrowserRouter>
        <Provider store={store}>
          <SignIn />
        </Provider>
      </BrowserRouter>
    )

    fireEvent.change(getByPlaceholderText('E-mail'), {
      target: { value: 'any_email@mail.com' }
    })
    fireEvent.change(getByPlaceholderText('Senha'), {
      target: { value: 'any_password' }
    })

    expect(fireEvent.click(getByText('Entrar'))).toBeTruthy()
  })

  it('Should return error message if onSubmit was throw', () => {
    const { getByText, getByPlaceholderText } = render(
      <BrowserRouter>
        <Provider store={store}>
          <SignIn />
        </Provider>
      </BrowserRouter>
    )

    fireEvent.change(getByPlaceholderText('E-mail'), {
      target: { value: '' }
    })
    fireEvent.change(getByPlaceholderText('Senha'), {
      target: { value: '' }
    })

    expect(fireEvent.click(getByText('Entrar'))).toBeTruthy()
  })

  it('Should return call handleSubmit on pass valid credentials', () => {
    const { getByText, getByPlaceholderText } = render(
      <BrowserRouter>
        <Provider store={store}>
          <SignIn />
        </Provider>
      </BrowserRouter>
    )

    fireEvent.change(getByPlaceholderText('E-mail'), {
      target: { value: 'any_email@mail.com' }
    })
    fireEvent.change(getByPlaceholderText('Senha'), {
      target: { value: 'any_password' }
    })

    expect(fireEvent.click(getByText('Entrar'))).toBeTruthy()
  })

  it('Should return error message if onSubmit was throw', () => {
    const { getByText, getByPlaceholderText } = render(
      <BrowserRouter>
        <Provider store={store}>
          <SignIn />
        </Provider>
      </BrowserRouter>
    )

    fireEvent.change(getByPlaceholderText('E-mail'), {
      target: { value: '' }
    })
    fireEvent.change(getByPlaceholderText('Senha'), {
      target: { value: '' }
    })

    expect(fireEvent.click(getByText('Entrar'))).toBeTruthy()
  })
})
