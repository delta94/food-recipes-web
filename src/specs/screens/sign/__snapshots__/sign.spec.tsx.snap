// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SCREENS] sign Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnylx loKanH"
      >
        <div
          class="sc-gtssRu dCIcPW"
        >
          <div
            class="sc-dlnjPT dxXvdR"
          >
            <a
              href="/"
            >
              <img
                alt="Recipes"
              />
            </a>
            <form>
              <h1>
                Faça seu login
              </h1>
              <input
                name="email"
                placeholder="E-mail"
                value=""
              />
              <input
                name="password"
                placeholder="Senha"
                type="password"
                value=""
              />
              <button
                type="submit"
              >
                Entrar
              </button>
            </form>
            <a
              href="/signup"
            >
              Criar conta
            </a>
          </div>
        </div>
        <div
          class="sc-hKFyIo ipKnoK"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnylx loKanH"
    >
      <div
        class="sc-gtssRu dCIcPW"
      >
        <div
          class="sc-dlnjPT dxXvdR"
        >
          <a
            href="/"
          >
            <img
              alt="Recipes"
            />
          </a>
          <form>
            <h1>
              Faça seu login
            </h1>
            <input
              name="email"
              placeholder="E-mail"
              value=""
            />
            <input
              name="password"
              placeholder="Senha"
              type="password"
              value=""
            />
            <button
              type="submit"
            >
              Entrar
            </button>
          </form>
          <a
            href="/signup"
          >
            Criar conta
          </a>
        </div>
      </div>
      <div
        class="sc-hKFyIo ipKnoK"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
